src/rgw/rgw_op.cc

```diff
@@ -5013,6 +5013,13 @@ void RGWDeleteObj::execute(optional_yield y)
         etag = astate->attrset[RGW_ATTR_ETAG].to_str();
       }
 
+      // if the object does not exist, it is not allowed to create a delete_marker
+      if(s->bucket->get_info().versioning_enabled()&&!s->object->have_instance()){
+        rgw_bucket_olh_entry olh;
+        op_ret = store->getRados()->bi_get_olh(this,s->bucket->get_info(),s->object->get_obj(),&olh);
```

Comment:
I'm not sure what this check actually does.  could someone please explain what calling get_olh() on this object (without intent to use the result) accomplishes?

Reply:
This call is used to determine whether the object exists. If the object does not exist, the object's olh should not exist, then ret < 0